assertj-core/src/main/java/org/assertj/core/error/ShouldHaveAnnotations.java

```diff
@@ -31,13 +32,13 @@ public class ShouldHaveAnnotations extends BasicErrorMessageFactory {
    * @param missing missing annotations for this class
    * @return the created {@code ErrorMessageFactory}.
    */
-  public static ErrorMessageFactory shouldHaveAnnotations(Class<?> actual,
+  public static ErrorMessageFactory shouldHaveAnnotations(AnnotatedElement actual,
```

Comment:
The binary compatibility check fails on widening this static method from `Class<?>` to `AnnotatedElement`. Class implements this interface since Java 5. I'm not sure on requirements to keep backward compatible with pre Java 5 code.

Reply:
The main problem is not necessarily related to compatibility with Java < 5 but rather to the signature change of `shouldHaveAnnotations`, which can break code compiled with AssertJ <= 3.24.2.

Could you maybe overload it and deprecate the existing one?

Reply:
Done